<!-- üß™ RETO: Generar un n√∫mero aleatorio -->
<!-- üî• RETO 2: Encadenar dos tareas asincr√≥nicas
Enunciado:

Usa la misma funci√≥n obtenerNumero() como primer paso.

Luego, si el primer n√∫mero fue exitoso, espera otro n√∫mero aleatorio con otra llamada a obtenerNumero().

Si ambos n√∫meros pasan la condici√≥n (mayor o igual a 0.3), muestra:

‚úÖ Todo bien. N√∫meros: 0.74 y 0.82

Si cualquiera de los dos falla, muestra el error correspondiente. -->

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Reto: N√∫mero aleatorio</title>
</head>
<body>
    <h1>N√∫mero Aleatorio</h1>
    <button id="btn">Generar n√∫mero</button>
    <p id="resultado"></p>

    <script>
        function obtenerNumero() {
            return new Promise((resolve, reject) => {
                setTimeout(() => {
                    const numero = Math.random();
                    if (numero >= 0.3) {
                        resolve(numero.toFixed(2));
                    } else {
                        reject(`Mal, demasiado bajo.`);
                    }
                }, 2000);
            });
        }

        async function mostrarResultado() {
            const resultado = document.getElementById("resultado");
            resultado.textContent = "Espera...";

            try {
                const mensaje = await obtenerNumero();
                resultado.innerHTML = `Primer n√∫mero generado: ${mensaje}<br>Generando segundo n√∫mero...`;

                const mensaje2 = await obtenerNumero();
                resultado.innerHTML = `Bien. N√∫meros generados correctamente:<br> ${mensaje} y ${mensaje2};`
                
            } catch (error) {
                resultado.textContent = `Error: ${error}`;
            }
        }

        document.getElementById("btn").addEventListener('click', mostrarResultado)
    </script>
</body>
</html>